sdk-setup/src/mb2

```diff
@@ -1194,6 +1200,60 @@ END
     chmod +x "$wrapper"
 )
 
+# 1. Allows to pass extra arguments to cmake
+# 2. Enables shadow build by adding path prefix to the project file, or
+# 2.1 passing project directory path if the project file was not passed
+init_cmake_wrapper() (
+    local wrapper=$1
+    local extra_args=("${@:2}")
+
+    main() {
+        self=${0#$PWD/}
+        command=${0##*/}
+        NBSP=$'\xC2\xA0'
+
+        path=${0%/*}
+        PATH=${PATH/$path:/}
+
+        real=$(which "$command")
+        if [[ ! $real || $real == "$0" ]]; then
+            echo "$self: Cannot find real $command" >&2
+            exit 1
+        fi
+
+        if [[ $IS_SHADOW_BUILD ]]; then
+            subdir=${PWD#$BUILD_DIR}
+            src_dir=$SRC_DIR${subdir:+/$subdir}
+            args=()
+            for arg in "$@"; do
+                args+=("$arg")
+            done
+            args+=("$src_dir")
+            set -- "${args[@]}"
```

Comment:
Compared to qmake case we do not do any $arg manipulation and append $src_dir unconditionally here. Is this really/still correct? If so, then I would simply replace this with `set -- "$@" "$src_dir"`.